pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -321,6 +357,16 @@ public void start()
     LOGGER.info("Initializing server instance and registering state model factory");
     Utils.logVersions();
     ControllerLeaderLocator.create(_helixManager);
+
+    // Invoke pinot environment provider factory's init method to register the environment provider
+    // and fetch the overridden server configs for the invoked environment provider
+    if (!skipPinotEnvironmentProviderFactoryInit) {
+      PinotEnvironmentProviderFactory.init(_serverConf.subset(PREFIX_OF_CONFIG_OF_ENVIRONMENT_PROVIDER_FACTORY));
+      PinotEnvironmentProvider pinotEnvironmentProvider = PinotEnvironmentProviderFactory.getEnvironmentProvider(
+          _serverConf.getProperty(CONFIG_OF_ENVIRONMENT_PROVIDER_TYPE));
+      _overriddenServerConfigs = pinotEnvironmentProvider.getEnvironment(_serverConf.toMap());
```

Comment:
why have overridden configs different from server config ?
it is "overidden" with the values appropriate to environment.
this is the new server config now.

Reply:
Done.